packages/suite/src/actions/suite/metadataActions.ts

```diff
@@ -946,3 +975,297 @@ export const setEditing = (payload: string | undefined): MetadataAction => ({
+        //    this means that they have already been migrated or dummy file was created
+        const everyEntityHasNewFile = dispatch(getLabelableEntities(device.state)).every(entity => {
+            const nextKeys = entity[METADATA.ENCRYPTION_VERSION];
+
+            if (!nextKeys) {
+                // todo: should never happend but happend once during testing.
+                throw new Error('metadata migration: next keys are missing');
+            }
+            return currentEncryptionFiles.find(file => file === nextKeys.fileName);
+        });
+
+        if (everyEntityHasNewFile) {
+            return { success: true };
+        }
+
+        // 7. sync metadata keys for prev encryption version.
+        //    NOTE: result of this operation is saved for device and account encryption keys are computed from it. This means that we can add new accounts at any point of time later without calling this again
+        if (!device.metadata[prevEncryptionVersion]) {
+            await dispatch(setDeviceMetadataKey(prevEncryptionVersion));
+        }
+        device = selectDevice(getState());
+        if (!device?.metadata[prevEncryptionVersion]?.key) {
+            return { success: false, error: 'metadata migration: cancelled' };
+        }
+
+        dispatch(syncMetadataKeys(prevEncryptionVersion));
+
+        if (!device?.state) {
+            // this should never happen
+            return { success: false, error: 'metadata migration: device not authorized' };
+        }
+
+        // 8. get labelable entitites again (there was async operation in between)
+        const allEntities = dispatch(getLabelableEntities(device.state));
+
+        // 9. split labelable entities into 2 groups:
+        //    - entitiesToMigrate: don't have new file && have old file
+        //    - entititiesToCreateDummies: don't have new file && don't have old file
+
+        const entitiesToMigrate: LabelableEntity[] = [];
+        const entititiesToCreateDummies: LabelableEntity[] = [];
+        allEntities.forEach(entity => {
+            const prevKeys = entity[prevEncryptionVersion];
+
+            if (!prevKeys) {
+                console.error('metadata migration: prev keys are missing');
+                return; // should never happen
+            }
+            const nextKeys = entity[METADATA.ENCRYPTION_VERSION];
+
+            if (!nextKeys) {
+                console.error('metadata migration: next keys are missing');
+                return; // should never happen
+            }
+
+            const oldFileExists = oldEncryptionFiles.find(file => file === prevKeys.fileName);
+            const newFileExists = currentEncryptionFiles.find(file => file === nextKeys.fileName);
+
+            if (newFileExists) {
+                return; // already migrated
+            }
+
+            if (!oldFileExists) {
+                entititiesToCreateDummies.push(entity);
```

Comment:
maybe I should be creating dummies only if I know I am doing "partial migration". this means that I know that there are more old files than labelable entities I have discovered

Reply:
probably not worth doing it. my biggest concern here is that at one point I would decide that I am not creating dummies, than during the course of migration migrating the last entity fails, but I don't have dummies thus I am likely to bother user next time. In general, this is also a product question - are we actually allowed to create dummies?

Reply:
I believe it's fine as it is.

Reply:
ok, just added reference to previous file so that we can handle future fuckups ebb92872b1aecbb5f53e156741fc6fba5767a17b